Source/CrossingDetector/CrossingDetector.cpp

```diff
@@ -341,26 +294,36 @@ void CrossingDetector::setParameter(int parameterIndex, float newValue)
         break;
 
     case pEventChan:
-        // if we're in the middle of an event, keep track of the old channel until it's done.
-        if (sampsToShutoff > -1)
-            shutoffChan = eventChan;
         eventChan = static_cast<int>(newValue);
         break;
 
     case pEventDur:
         eventDuration = static_cast<int>(newValue);
+        if (CoreServices::getAcquisitionStatus())
+        {
+            float sampleRate = getDataChannel(inputChan)->getSampleRate();
+            eventDurationSamp = static_cast<int>(ceil(eventDuration * sampleRate / 1000.0f));
+        }
         break;
 
     case pTimeout:
         timeout = static_cast<int>(newValue);
+        if (CoreServices::getAcquisitionStatus())
+        {
+            float sampleRate = getDataChannel(inputChan)->getSampleRate();
+            timeoutSamp = static_cast<int>(floor(timeout * sampleRate / 1000.0f));
+        }
         break;
 
     case pPastSpan:
         pastSpan = static_cast<int>(newValue);
-        sampsToReenable = pastSpan + futureSpan + 1;
+        sampToReenable = pastSpan + futureSpan + 1;
         pastBinary.resize(pastSpan + 1);
         pastBinary.clearQuick( );
         pastBinary.insertMultiple(0, 0, pastSpan + 1);
+        pastCounter = 0; // must reflect current contents of pastBinary
```

Comment:
There was a tricky bug where `pastCounter` and `futureCounter` were getting off after stopping and restarting acquisition. I realized this was because I had added lines to reset them to 0 when acquisition stopped, which didn't actually reflect the contents of the buffers, since I wasn't clearing them. Then when the old data points equaling 1 left the array, `pastCounter` and `futureCounter` went negative. So my conclusion was that it's important that `pastCounter` and `futureCounter` always reflect the number of 1's in their corresponding arrays, even if we don't care about the data in the arrays.